### PR TITLE
Reauth Endpoint

### PR DESCRIPTION
**Pull request by alfmags**

### CODE REVIEW

**Comment by cloveras** [comment]
Should `paymentInstrument` be an enum instead of just a string? Will it be a problem if we get `VISA`, `Visa`, `visa`, `MasterCard`, `Mastercard`, `mastercard`, etc?

File: docs/swagger.yaml [hunk 1]
```diff
@@ -2144,6 +2300,13 @@ components:
           $ref: '#/components/schemas/MerchantInfoPayment'
         transaction:
           $ref: '#/components/schemas/CancelTransaction'
+    AuthroizePaymentRequest:
+      type: object
+      properties:
+        merchantInfo:
+          $ref: '#/components/schemas/MerchantInfoPayment'
+        transaction:
+          $ref: '#/components/schemas/CancelTransaction'
```
Comment (tht13) [inline docs/swagger.yaml:2309]: Maybe this needs renaming now, how about `SimpleTransaction` and `ValueTransaction` for transaction objects with and without a "amount" property?

**Comment by alfmags** [comment]
> Should `paymentInstrument` be an enum instead of just a string? Will it be a problem if we get `VISA`, `Visa`, `visa`, `MasterCard`, `Mastercard`, `mastercard`, etc?

The problem with it being an ENUM is that we don't know the full set yet of output, we might be adding "Account" later...

**Comment by tht13** [comment]
> > Should `paymentInstrument` be an enum instead of just a string? Will it be a problem if we get `VISA`, `Visa`, `visa`, `MasterCard`, `Mastercard`, `mastercard`, etc?
> 
> The problem with it being an ENUM is that we don't know the full set yet of output, we might be adding "Account" later...

@cloveras this is also a field in a response object, so we will dictate the casing here (likely all uppercase), inputs will obviously be a enum to make values constant.

**Comment by alfmags** [comment]
> Should `paymentInstrument` be an enum instead of just a string? Will it be a problem if we get `VISA`, `Visa`, `visa`, `MasterCard`, `Mastercard`, `mastercard`, etc?

Since we are intending to add future instruments I think string is better, but with clear examples of what to expect. We will use an enum internally.

File: docs/swagger.yaml (hunk 1 above)
Comment (alfmags) [reply docs/swagger.yaml:2309]: Let's take that in a separate PR, not sure if I agree.

**Pull request merged**